src/codeu/chat/client/ClientUser.java

```diff
@@ -49,10 +49,11 @@ static public boolean isValidName(String userName) {
     boolean clean = true;
     if (userName.length() == 0) {
       clean = false;
+    //if alphanumeric
+    } else if(!userName.matches("[a-zA-Z0-9]+")) {
```

Comment:
Don't you guys already have a compiled Pattern somewhere with this username pattern? Consolidating the pattern in one location will allow you to seamlessly update if you decide to allow more interesting characters,